### DISCUSSION TITLE
Notifications generate too many Debug Output messages.

### DISCUSSION THREAD

**Discussion by Glomby** [shinyorg/shiny#1451] [Q&A] [answered]
I’m upgrading an xamarin project to Maui right now and am running into a weird problem. I was using an older version of Shiny in Xamarin but also upgraded to the newest release for Maui.

One big difference right now seems to be that Shiny provides a lot more debug information in the output console, which is obviously great but it also shows output for each received BLE Notification. The device I’m developing for sometimes (especially after the first connection) sends notifications every 15 or 7,5ms. This basically spams all my output window and even worse slows down the entire app when in debug mode.

is it possible to adjust what debug messages are generated? Like an opt-in opt-out? Or is there something else I could do?

**Comment by aritchie** [comment, marked as answer]
https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#configure-logging

**Reply by Glomby** [reply]
Thank you very much. Works as expected. If anyone is finding this the LogLevel I needed to turn down was for the IPeripheral.

Like this:
`.AddFilter(“Shiny.BluetoothLE.IPeripheral”, LogLevel.Warning);`

**Reply by aritchie** [reply]
https://github.com/shinyorg/shiny/blob/a3458bf5cbb86c9f0321f582d1bed6b0d3188573/samples/Sample.Maui/MauiProgram.cs#L36C9-L36C57
